gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -166,7 +166,11 @@ namespace gtsam {
   DiscreteKeys DecisionTreeFactor::discreteKeys() const {
     DiscreteKeys result;
     for (auto&& key : keys()) {
-        result.emplace_back(key, cardinality(key));
+        DiscreteKey discreteKey(key, cardinality(key));
+        // Only add unique keys
+        if (std::find(result.begin(), result.end(), discreteKey) == result.end()) {
```

Comment:
wait this is a bug? Why will there be non-unique keys?

Reply:
Hmm yeah maybe not. I don't remember why I did this, but if it crops up again, at least I'll know why.

Reply:
Removed

Reply:
There are other places with this check. I propose that we should remove them.